crates/svm-cli/src/spawn_app.rs

```diff
@@ -0,0 +1,120 @@
+use std::convert::TryFrom;
+use std::error::Error;
+use std::fs::File;
+use std::io::prelude::*;
+
+use svm_app::{
+    raw::{decode_spawn_app, NibbleIter},
+    testing::SpawnAppBuilder,
+    types::{SpawnApp, WasmValue},
+};
+use svm_common::Address;
+
+pub fn encode(
+    version: u32,
+    template_addr_hex: &str,
+    ctor_idx: u16,
+    ctor_buf_hex: &str,
+    ctor_args: Option<Vec<&str>>,
+    output_path: &str,
+) -> Result<usize, Box<dyn Error>> {
+    let template_addr = match hex::decode(template_addr_hex) {
+        Ok(v) => v,
+        Err(e) => {
+            let e = format!(
+                "failed to decode hex string '{}': {}",
+                template_addr_hex,
+                e.to_string()
+            );
+            return Err(e.into());
+        }
+    };
+    let template_addr = {
+        if Address::len() != template_addr.len() {
+            return Err(format!(
+                "invalid template address length, found {}, expected: {}",
+                template_addr.len(),
+                Address::len()
+            )
+            .into());
+        }
+        Address::from(template_addr.as_slice())
+    };
+
+    let ctor_buf = match hex::decode(ctor_buf_hex) {
+        Ok(v) => v,
+        Err(e) => {
+            let e = format!(
+                "failed to decode hex string '{}': {}",
+                ctor_buf_hex,
+                e.to_string()
+            );
+            return Err(e.into());
+        }
+    };
+
+    let mut ctor_args_vals: Option<Vec<WasmValue>> = None;
```

Comment:
please rename it as `ctor_args`.

Reply:
A different name is needed since i'm using `Vec<&str>` to populate `Vec<WasmValue>`.